### ISSUE TITLE
Invalid CloudRail license keys

### ISSUE THREAD

**Issue by p0087** [CloudRail/cloudrail-si-android-sdk#73]
Hi,
I'm not sure what causes this, nor can I reproduce this myself, but I've had a couple of reports containing: 

```
java.lang.RuntimeException: A valid CloudRail license key is required. You can get one for free at https://developers.cloudrail.com
at com.cloudrail.si.CloudRail.validateKey(CloudRail.java)
at com.cloudrail.si.services.GoogleDrive.upload(GoogleDrive.java)
```

I've got a license key and set `CloudRail.setAppKey("blah");`, and it's working fine for 99.99% of people. Just wondering if this something possibly on your end going wrong.

**Comment by PatrickStoklasa** [comment]
Is the line where you set the app key maybe not executed in some cases? The error is really only thrown if the app key wasn't set correctly or it is in the wrong format (if the key has the correct format but is invalid you would get a different error message). So the only thing I could imagine is that there might be some circumstances where the line is not executed.

**Comment by p0087** [comment]
Yeah it's called in both onCreate and onResume before any attempts to upload. Barring GC, I don't see what could cause it either. I've only received a couple reports about it, so it's not a big issue, just thought it might be an internal issue.

**Comment by PatrickStoklasa** [comment]
Ok then, I will keep it here and see if anyone else faces the same issue. If you find a way to reproduce it in the meantime just let me know.
